File: commons-text/src/test/java/test/pl/wrzasq/commons/text/slugifier/SlugifierTest.java
/*
 * This file is part of the pl.wrzasq.commons.
 *
 * @license http://mit-license.org/ The MIT license
 * @copyright 2015, 2018 - 2019 © by Rafał Wrzeszcz - Wrzasq.pl.
 */

package test.pl.wrzasq.commons.text.slugifier;

import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;
import pl.wrzasq.commons.text.slugifier.Slugifier;

public class SlugifierTest
{
    @Test
    public void slugify()
    {
        Slugifier slugifier = new Slugifier()
        {
            @Override
            public String slugify(String... texts)
            {
                return "foo";
            }
        };

        Assertions.assertEquals(
            "foo",
            slugifier.slugify("test"),
            "Slugifier.slugify() should by default apply slugifier logic on sinle-elemene array."
        );
    }
}

File: commons-text/src/test/kotlin/test/pl/wrzasq/commons/text/slugifier/SlugifierTest.kt
/*
 * This file is part of the pl.wrzasq.commons.
 *
 * @license http://mit-license.org/ The MIT license
 * @copyright 2015, 2018 - 2019, 2021 © by Rafał Wrzeszcz - Wrzasq.pl.
 */

package test.pl.wrzasq.commons.text.slugifier

import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Test
import pl.wrzasq.commons.text.slugifier.Slugifier

class SlugifierTest {
    @Test
    fun slugify() {
        val slugifier = Slugifier { "foo" }
        assertEquals("foo", slugifier.slugify("test"))
    }
}
